src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -192,10 +193,15 @@ updateAddrBalances (combineBalanceUpdates -> updates) = mapM_ updater updates
     updater (addr, (Minus, coin)) = do
         maybeBalance <- getAddrBalance addr
         case maybeBalance of
-            Nothing -> logError "updateAddrBalances: subtracting coins from unknown address"
+            Nothing ->
+                logError $
+                    sformat ("updateAddrBalances: attempted to subtract "%build%" coins from unknown address "%build)
+                    coin addr
             Just currentBalance
                 | currentBalance < coin ->
-                    logError "updateAddrBalances: address does not have enough coins"
+                    logError $
+                        sformat ("updateAddrBalances: attempted to subtract "%build%" coins from address "%build%" which only has "%build%" coins")
```

Comment:
This line seems to be extremely long 😔 

Also, coin is formatted as `5 coin(s)`. If you write `coins` in format, then use `int` formatter (which will format it as `5`).